### ISSUE TITLE
Allow HTTP requests to *.stripe.com

### ISSUE THREAD

**Issue by cagataygurturk** [stripe/stripe-node#2006]
### Is your feature request related to a problem? Please describe.

Currently the library does not allow HTTP connections to Stripe API, rightfully. However, changing the protocol back to HTTP might be required in niche scenarios such as using Istio's [Egress TLS Origination](https://istio.io/latest/docs/tasks/traffic-management/egress/egress-tls-origination/) feature. In this case, the TLS encryption is done by the sidecar container, not by the application and Istio can perform advanced traffic management towards Stripe API.

### Describe the solution you'd like

Either completely remove the following block or add a property to turn off the check.

https://github.com/stripe/stripe-node/blob/3fee1aac5070da92ba0f0bfe48e6396834704cd1/src/stripe.core.ts#L107-L115

In fact, the library already allows setting protocol to `http`, so one can think that such a property already exists.

### Additional context

Stripe API anyways is not allowing HTTP connections and accidentally turning on HTTP connections will not create a harm.

````bash
$ curl http://api.stripe.com
````
````json
{
  "error": {
    "message" : "The Stripe API is only accessible over HTTPS.  Please see <https://stripe.com/docs> for more information.",
    "type": "invalid_request_error"
  }
}
````

**Comment by richardm-stripe** [comment]
Hello @cagataygurturk. Thanks for the report. Your requests makes sense, thanks for including specific details about istio. We have removed the check in #2009 which should be included in [v14.15.0](https://github.com/stripe/stripe-node/releases/tag/v14.15.0) due for release later this week.

**Comment by anniel-stripe** [comment]
Hi! The change in https://github.com/stripe/stripe-node/pull/2009 was released in [v14.15.0](https://github.com/stripe/stripe-node/releases/tag/v14.15.0) today. Closing out this issue.
